src/c-writer.cc

```diff
@@ -2645,16 +2847,38 @@ void CWriter::WriteUnwindTryCatchStack(const Label* label) {
   }
 }
 
-void CWriter::Write(const Func& func) {
+void CWriter::FinishReturnCall() {
+  if (in_tail_callee_) {
+    Write(CloseBrace(), Newline(), "return;", Newline());
+    return;
+  }
+
+  Write("while (next->fn) { next->fn(instance_ptr, tail_call_stack, next); }",
+        Newline());
+  PushTypes(func_->decl.sig.result_types);
+  const Index num_results = func_->decl.sig.result_types.size();
+  if (num_results == 1) {
```

Comment:
Code cleanup question: Can this part of the code where c-writer handle single return vs. multi-value return be factored out into a helper that can be used in the "ExprType::Call"" handling as well as here? It seems a bit messy having to special case this multiple times. (If not, no worries. Not blocking)

Reply:
Yeah, agreed. I think we did this partway with Spill/Unspill, but I think if you look at all the places we have an `if (num_params == 1)` and `if (num_results == 1)` in the code, I couldn't find enough commonality to factor out further. :-/